### ISSUE TITLE
ContextualMenu A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#6360]
Tracking all of the issues brought up with the 08/24 accessibility pass on Contextual Menu  control
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Bcef6201c-bd85-44b6-a2d3-3784248461e0%7D&action=default&uid=%7BCEF6201C-BD85-44B6-A2D3-3784248461E0%7D&ListItemId=1259&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- [x] Bug 1 : [Accessibility][Forms and validation]: 'Actions' label is not associated with its corresponding control.
  - @kevintcoughlin: This is a header node, so perhaps the correct fix here is to use `aria-label` on the search input as that element is missing a corresponding label.
- [x] Bug 2 : [Accessibility][Screen reader]: Label or place holder is not provided for the Edit field(250) present under Contextual Menu with Sub Menus
- [ ] Bug 3 : [Accessibility][Screen reader]: Narrator focus moves twice on the first element of context menu in scan mode.
- [ ] Bug 4 : [Accessibility][Screen reader]: Name, Role and state is not properly defined for split buttons under 'ContextualMenu with checkable menu items and toggleable split button'.
- [x] Bug 5 : [Accessibility][Screen reader]: Secondary text is not announced when user navigates in context menu under ContextualMenu with icons and secondary text [#6670].  
- [ ] Bug 6 : [Accessibility][Screen reader]: Narrator does not read notification alert properly.
- [ ] Bug 7 : [Accessibility][Screen reader]: In Scan mode user is not able to close the 'Contextual menu' using 'Esc key'.
  - @KevinTCoughlin: This is likely because scan focuses different elements and therefore the event isn't bubbling on the component.
- [ ] Bug 8 : [Accessibility][Keyboard Navigation]: Tab Focus is moving out of the Notification window which appears on activating 'Share w/split button'.
- [x] Bug 9 : [Accessibility][Screen reader]: Narrator focus moves onto the separator line in scan mode. [#6876]
- [ ] Bug 10 : [Accessibility][Screen reader]: Narrator focus moves onto hidden elements  in scan mode in backward navigation.
- [ ] Bug 11 : [Accessibility][Screen reader]: User is not able to access "Share w/split button " in scan mode.
- [x] Bug 12 : [Accessibility][Programmatic access]: id attribute value must be unique.
- [ ] Bug 13 : [Accessibility][Programmatic access]: 'Hierarchy levels' are not defined for submenus.   
- [x] Bug 14 : [Functional]: Some controls are not implemented properly in 'Charm' submenu window.
- [ ] Bug 15 : [Accessibility][Usable]: Header/Label should be define for 'Categories group' in 'Categorize submenu'.
  - @KevinTCoughlin: Not sure if we can do this with current API. `ariaLabel` on `subMenuProps` adds the `aria-label` to the button, not the callout submenu.
- [ ] Bug 16 : [Accessibility][Programmatic access]: Role of the categories is not appropriate as multiple selection of categories could be possible.  
- [ ] Bug 17 : [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation.
- [x] Bug 18 : [Accessibility][Keyboard]: User is not able to delete 'NAN' value from 'Hover delay' edit box under 'ContextualMenu with submenus' after entering invalid values.

Component: ContextualMenu

@micahgodbolt

**Comment by KevinTCoughlin** [comment]
Crossing off bugs that I've committed fixes for in https://github.com/KevinTCoughlin/office-ui-fabric-react/tree/keco/ctx-menu-demo-a11y as I go.

**Comment by micahgodbolt** [comment]
@KevinTCoughlin careful not to say "fixes" this issue, unless it closes all the listed bugs.

**Comment by dzearing** [comment]
@joschect ping.
